File: common/src/main/java/org/sandboxpowered/silica/network/handshake/serverbound/PongResponse.java
package org.sandboxpowered.silica.network.handshake.serverbound;

import org.sandboxpowered.silica.network.*;

public class PongResponse implements Packet {
    private long time;

    public PongResponse(long time) {
        this.time = time;
    }

    public PongResponse() {
    }

    @Override
    public void read(PacketByteBuf buf) {
        this.time = buf.readLong();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeLong(this.time);
    }

    @Override
    public void handle(PacketHandler packetHandler, Connection connection) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/handshake/serverbound/PongResponse.kt
package org.sandboxpowered.silica.network.handshake.serverbound

import org.sandboxpowered.silica.network.Connection
import org.sandboxpowered.silica.network.Packet
import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler

class PongResponse(private var time: Long) : Packet {

    constructor() : this(0)

    override fun read(buf: PacketByteBuf) {
        time = buf.readLong()
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeLong(time)
    }

    override fun handle(packetHandler: PacketHandler, connection: Connection) {}
}